### PR TITLE
chore(#690): Setup Code Scanning

### LINKED ISSUE #690: Implement Code Scan for the repo
## Is your chore related to a problem? Please describe
We have access to GitHub's Code Scan functionality and as such, we need to set up code scanning configuration to activate Code Scanning for the repo.
 
## Describe the solution you'd like
Setup the CodeQL workflow to enable Code Scanning

### PR DESCRIPTION
**Pull request by sayak-sarkar**
# Closes/Fixes/Resolves
#690 

# Explain the feature/fix
Setup the CodeQL Analysis workflow for GitHub to enable Code Scanning for the repo.

## Does this PR introduce a breaking change
No

### Ready-for-merge Checklist
- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Did tests pass?

### CODE REVIEW

**Review by riginoommen — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['javascript']
```
Comment (deshmukhmayur) [inline .github/workflows/codeql-analysis.yml:22]: Can we also add typescript to it?

Since out microservices are written in typescript.

Comment (sayak-sarkar) [reply .github/workflows/codeql-analysis.yml:22]: For the current Beta release, only JavaScript is supported. I'll add Typescript once it becomes available for code scanning.

Comment (sayak-sarkar) [reply .github/workflows/codeql-analysis.yml:22]: Later I am planning to look into https://github.com/SonarSource/SonarJS

**Review by ghanlohar — APPROVED** [review]
Looks good to me.

**Pull request merged**